### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by ab-noori**
### 🌟Features:

> - [x] - Following validations for the User model have been added.
              👉  The name must not be blank.
               👉 PostsCounter must be an integer greater than or equal to zero.
> - [x] - Following validations for the Post model have been added.
                👉  Title must not be blank.
                👉  Title must not exceed 250 characters.
                👉  CommentsCounter must be an integer greater than or equal to zero.
                👉  LikesCounter must be an integer greater than or equal to zero.
> - [x] - Unit specs for all models' methods and validations have been added

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
```
Comment (Owusu-Desmond) [inline spec/models/post_spec.rb:3]: - [x] I suggest that any custom methods in the models should have a corresponding unit test. So make sure that you add a test for the methods `update_posts_counter` and `recent_comments` in your Post model. :blush:

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  it 'is not valid without a post' do
+    subject.post = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'increments the likes_counter of the associated post after saving' do
+    expect { subject.save }.to change { post.reload.likes_counter }.by(1)
+  end
```
Comment (Owusu-Desmond) [inline spec/models/like_spec.rb:33]: - [x] - [ ] I suggest that any custom methods in the models should have a corresponding unit test. So make sure that you add a test for the method `update_post_likes_counter` in your Like model. :blush:

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
```
Comment (Owusu-Desmond) [inline spec/models/comment_spec.rb:3]: - [x] I suggest that any custom methods in the models should have a corresponding unit test. So make sure that you add a test for the method `update_post_comments_counter` in your Comment model. :blush:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @ab-noori 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Your added tests pass✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @Owusu-Desmond  👋,**
Thank you so much 🙏 for reviewing my pull request.
Your nice feedback is really appreciated 🙏.
I have checked and made them fix 🎯.

Thank-You-Very-Much

**Review by franclobo — APPROVED** [review]
Hi @ab-noori, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Use validations for models. ✔️ 
- Write unit tests for models. ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @franclobo 👋,**
Thank you so much 🙏 for reviewing and approving my pull request.
Your nice feedback is really appreciated 🙏.

thank-you-gif-22

**Pull request merged**